internal/controllers/gateway/gateway_utils.go

```diff
@@ -217,20 +217,20 @@ func initializeListenerMaps(gateway *Gateway) (
 func canSharePort(requested, existing ProtocolType) bool {
 	switch requested {
 	// TCP and UDP listeners must always use unique ports
-	case (ProtocolType)(gatewayv1alpha2.TCPProtocolType), (ProtocolType)(gatewayv1alpha2.UDPProtocolType):
+	case gatewayv1beta1.TCPProtocolType, gatewayv1beta1.UDPProtocolType:
```

Comment:
In several places like these, I believe we can already start using the type aliases that we have in https://github.com/Kong/kubernetes-ingress-controller/blob/5f9f569379ed5af8d820efdae2e236a65076bc53/internal/controllers/gateway/types.go#L33-L39. WDYT?

Reply:
Honestly, I don't get the usefulness of such aliases 🤔 can you enlighten me?

Reply:
In a PR like this where we bump a version and/or an object type gets promoted from e.g. alpha to beta you change only that 1 place

Reply:
I started refactoring, then I realized that the current place for those types is not optimal (it is under `internal/controllers`, and importing the controllers' package seems weird to me and can cause cyclic dependencies). I think we should move it to the types package we already have. Let's dedicate an entire PR to moving the package elsewhere and using it consistently across our code base. WDYT, if I create a dedicated issue for this and leave this PR as it is?

Reply:
Agreed 💯 . I also noticed that but never got around to it to change it.

> WDYT, if I create a dedicated issue for this and leave this PR as it is?

👍

Reply:
https://github.com/Kong/kubernetes-ingress-controller/issues/3503